Tools/CISupport/ews-build/steps.py

```diff
@@ -7352,6 +7352,95 @@ def getResultSummary(self):
         return {u'step': status}
 
 
+class DownloadUnexpectedResultsFromMaster(transfer.FileDownload):
+    mastersrc = WithProperties('public_html/results/%(buildername)s')
+    workerdest = SCAN_BUILD_OUTPUT_DIR + '/unexpected_results.json'
+    name = 'download-unexpected-results-from-master'
+    description = ['downloading unexpected results from buildbot master']
+    descriptionDone = ['Downloaded unexpected results']
+    haltOnFailure = True
+    flunkOnFailure = True
+    test = False
+
+    def __init__(self, **kwargs):
+        # Allow the unit test to override mastersrc
+        if 'mastersrc' not in kwargs:
+            kwargs['mastersrc'] = self.mastersrc
```

Comment:
I took these from `DownloadBuiltProductFromMaster` - we could probably operate without them given the file is small, but I thought it would be good to be explicit, especially with the permissions (`mode=644`)